lib/src/queue_list.dart

```diff
@@ -261,7 +261,9 @@ class QueueList<E> extends Object with ListMixin<E> implements Queue<E> {
 class _CastQueueList<S, T> extends QueueList<T> {
   final QueueList<S> _delegate;
 
-  _CastQueueList(this._delegate) : super._() {
+  /// Assigns invalid values for head/tail because it uses the delegate to hold
+  /// the real values, but they are non-null fields.
+  _CastQueueList(this._delegate) : super._(-1, -1) {
     _table = _delegate._table.cast<T>();
   }
```

Comment:
I'm pretty sure we can access `_delegate` in the initializer list now.
```suggestion
  _CastQueueList(this._delegate)
      : _table = _delegate._table.cast<T>(),
        super._(-1, -1);
```

Reply:
Unfortunately we can't - `_table` is actually from the super class (🤢)

Reply:
We can pass it through the `super._` call then. I suspect we could even make it final and initialize it in the initializer list in the parent class. If you want to file an issue I can follow up on this outside of this PR.

Reply:
done